raft/raft.go

```diff
@@ -718,14 +717,10 @@ func stepLeader(r *raft, m pb.Message) {
 				return
 			}
 			r.abortLeaderTransfer()
-			r.logger.Infof("%x [term %d] abort transfer leadership to %x", r.id, r.Term, lastLeadTransferee)
+			r.logger.Infof("%x [term %d] abort previous transferring leadership to %x", r.id, r.Term, lastLeadTransferee)
 		}
 		if leadTransferee == r.id {
-			if lastLeadTransferee == None {
-				r.logger.Debugf("%x is already leader. Ignored transfer leadership to %x", r.id, r.id)
-			} else {
-				r.logger.Debugf("%x abort transfer leadership to %x, transfer to current leader %x.", r.id, lastLeadTransferee, r.id)
-			}
+			r.logger.Debugf("%x is already leader. Ignored transferring leadership to self", r.id)
```

Comment:
It should already be reset in the if case at line713. That handles when the lastLeadTransferee is not None and abort it when necessary. When we are at line 721, leadTransferee should be None since we either abort it at line 719 or we never go into the if statement since lastLeadTransferee is already None.

Line 722 is to ensure we never try to transfer leadership to leader itself.